### PR TITLE
Split up the action workflow, and hardened error handling

### PR DESCRIPTION
**Pull request by rorychatterton**
### TL;DR

Refactored the action to improve error handling, and ensure that if an action is skipped, it is clear to the customer why it happened.

### What changed?
- Split the workflow logic into separate files: `config.ts`, `graphiteService.ts`, and `main.ts`
- Added TypeScript interfaces for `RequestBody` and `ResponseBody`
- Added `core.info('Skip: xxx')` to functions to clarify what the action did (as it is not easy to see outside debug mode by default).
- Added validation for PR Number, to skip in circumstances the action is unrelated to a PR (such as running on main). 
- Removed redundant `github_token` from action input
- Added Timeout Handling for the Graphite Service Request using controller signals & set max timeout to 5m.
- Updated README to reflect changes in action usage

### CODE REVIEW

**Comment by rorychatterton** [comment]
* **#2** <a href="https://app.graphite.dev/github/pr/rorychatterton/graphite-ci-action/2?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#1** <a href="https://app.graphite.dev/github/pr/rorychatterton/graphite-ci-action/1?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @rorychatterton and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>
